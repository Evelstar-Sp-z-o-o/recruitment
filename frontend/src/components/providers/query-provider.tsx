import { FC } from 'react';

import { QueryClientProvider, QueryClient } from '@tanstack/react-query';

const queryClient = new QueryClient();

interface Props {
  children: React.ReactNode;
}

const QueryProvider: FC<Props> = ({ children }) => {
  return <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>;
};

export default QueryProvider;
