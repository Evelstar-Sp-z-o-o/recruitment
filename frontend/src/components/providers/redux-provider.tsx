import store from "@/src/redux/store";
import { Provider } from "react-redux";

interface Props {
    children: React.ReactNode;
}

const ReduxProvider: React.FC<Props> = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
};

export default ReduxProvider;
