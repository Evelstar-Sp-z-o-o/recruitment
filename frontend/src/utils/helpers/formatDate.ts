export const formatDate = (date) => {
  const dateToConvert = new Date(date);

  return `${dateToConvert.getDate()}.${
    dateToConvert.getMonth() && dateToConvert.getMonth() + 1 > 9
      ? dateToConvert.getMonth() + 1
      : `0${dateToConvert.getMonth() + 1}`
  }.${dateToConvert.getFullYear()}, ${dateToConvert.getHours()}:${
    dateToConvert.getMinutes() && dateToConvert.getMinutes() > 9
      ? dateToConvert.getMinutes()
      : `0${dateToConvert.getMinutes()}`
  }`;
};
