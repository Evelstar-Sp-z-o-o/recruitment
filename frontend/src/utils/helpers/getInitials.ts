export const getInitials = (user: string) => {
  const initials: string[] = [];
  if (user.length > 0) {
    const name = user.substring(0, user.indexOf('@'));
    for (let i = 0; i < name.split('.').length; i++) {
      initials.push(name.split('.')[i][0].toUpperCase());
    }
  }

  return initials.join('');
};
