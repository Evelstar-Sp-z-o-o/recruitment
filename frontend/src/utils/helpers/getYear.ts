export const getYear = () => {
  const currentDate = new Date();
  return currentDate.getFullYear();
};
