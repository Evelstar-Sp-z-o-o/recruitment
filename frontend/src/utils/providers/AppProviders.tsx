import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import i18next from 'i18next';

import { store } from '@/src/store';

const AppProviders = ({ children }) => {
  return (
    <I18nextProvider i18n={i18next}>
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    </I18nextProvider>
  );
};

export default AppProviders;
