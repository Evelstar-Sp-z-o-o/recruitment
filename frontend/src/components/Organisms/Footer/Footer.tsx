import { FC } from 'react';

import { getYear } from '@/src/utils/helpers/getYear';
import Container from '@mui/material/Container';
import Typography from '@mui/material/Typography';

const StyledFooter = {
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  p: '2rem',
};

const Footer: FC = () => {
  return (
    <Container component="footer" className="footer" sx={StyledFooter}>
      <Typography className="appName">TWIXER</Typography>
      <Typography className="copyrights">&copy; {getYear()}</Typography>
    </Container>
  );
};

export default Footer;
