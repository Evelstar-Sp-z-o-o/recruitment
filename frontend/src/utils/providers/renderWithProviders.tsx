import { JSX, PropsWithChildren } from 'react';

import { render } from '@testing-library/react';

import AppProviders from './AppProviders.tsx';

export const renderWithProviders = (children: PropsWithChildren<JSX.Element>) => {
  return render(<AppProviders>{children}</AppProviders>);
};
