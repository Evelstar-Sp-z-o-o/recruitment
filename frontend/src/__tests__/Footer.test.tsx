import Footer from '@/src/components/Organisms/Footer/Footer';
import { getYear } from '@/src/utils/helpers/getYear';
import { renderWithProviders } from '@/src/utils/providers/renderWithProviders';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

vi.mock('@/src/utils/helpers/getYear');

describe('Footer', () => {
  it('renders app name and current year', () => {
    (getYear as jest.Mock).mockReturnValue(2024);

    render(<Footer />);

    expect(screen.getByText('TWIXER')).toBeInTheDocument();
    expect(screen.getByText('Â© 2024')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const snapshot = renderWithProviders(<Footer />);
    expect(snapshot).toMatchSnapshot();
  });
});
