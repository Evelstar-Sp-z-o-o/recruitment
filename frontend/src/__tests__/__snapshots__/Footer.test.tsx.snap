// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="MuiContainer-root MuiContainer-maxWidthLg footer css-7f8y8b-MuiContainer-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 appName css-ahj2mt-MuiTypography-root"
        >
          TWIXER
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 copyrights css-ahj2mt-MuiTypography-root"
        >
          © 
          2024
        </p>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="MuiContainer-root MuiContainer-maxWidthLg footer css-7f8y8b-MuiContainer-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 appName css-ahj2mt-MuiTypography-root"
      >
        TWIXER
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 copyrights css-ahj2mt-MuiTypography-root"
      >
        © 
        2024
      </p>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
